apps/api/src/app/shared/helpers/content.service.ts

```diff
@@ -85,4 +86,18 @@ export class ContentService {
   private escapeForRegExp(content: string) {
     return content.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
   }
+
+  buildMessageVariables(command: TriggerEventCommand, subscriberPayload) {
+    const messageVariables: { [key: string]: any } = { ...command.payload };
+    this.combineObjects(messageVariables, subscriberPayload, 'subscriber');
+
+    return messageVariables;
+  }
+
+  combineObjects(messageVariables: { [p: string]: any }, subscriberPayload, subscriberString = '') {
+    Object.keys(subscriberPayload).map(function (key) {
+      const newKey = subscriberString === '' ? key : `${subscriberString}.${key}`;
+      messageVariables[newKey] = subscriberPayload[key];
+    });
```

Comment:
It's usually better to avoid reference overrides with javascript and keep the functions pure that return a new instance of the object rather than overriding it with a `void` signature.